src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -149,7 +149,44 @@ void InputModel::InputModelTFImpl::load_places() {
         all_op_names.insert(op_name);
         m_op_places.push_back(op_place);
         m_op_places_map[op_name] = op_place;
+
+        // compute non-terminating nodes in the graph
+        // and put such nodes into op_names_with_consumers
+        for (size_t input_port_idx = 0; input_port_idx < node_decoder->get_input_size(); ++input_port_idx) {
+            std::string producer_op_name;
+            std::string producer_output_port_name;
+            size_t producer_output_port_idx;
+            try {
+                node_decoder->get_input_node(input_port_idx,
+                                             producer_op_name,
+                                             producer_output_port_name,
+                                             producer_output_port_idx);
+                if (is_conditional_edge(producer_op_name)) {
+                    // exclude "^" mark indicating (execution) conditional dependency
+                    // for example, "^sub_op" means dependency on a producer node with a name "sub_op"
+                    // if a node has dependent operation nodes and has no data consumers,
+                    // this node is not terminating and will not output to the Result node
+                    producer_op_name = producer_op_name.substr(1);
```

Comment:
Are we sure we don't have similar name without ^? I mean is it possible we have two nodes with names "^sub_op" and regular "sub_op"?

Reply:
No, it is not possible and `^` is used as special symbol to mark conditional dependency.